cmd/dagger/main.go

```diff
@@ -78,12 +78,48 @@ var (
 	Frontend idtui.Frontend
 )
 
+var (
+	stdin          io.Reader
+	stdout, stderr io.Writer
+)
+
+func logFileFromEnv(envName string) *os.File {
+	if logFileName := os.Getenv(envName); logFileName != "" {
+		f, err := os.Create(logFileName)
+		if err != nil {
+			panic(err)
+		}
+		return f
+	}
+	return nil
+}
+
+func wrapReader(r io.Reader, envName string) io.Reader {
+	f := logFileFromEnv(envName)
+	if f == nil {
+		return r
+	}
+	return io.TeeReader(r, f)
+}
+
+func wrapWriter(w io.Writer, envName string) io.Writer {
```

Comment:
kinda nit, but iiuc these pass straight through if vars are unset, right?

```suggestion
func maybeWrapReader(r io.Reader, envName string) io.Reader {
	f := logFileFromEnv(envName)
	if f == nil {
		return r
	}
	return io.TeeReader(r, f)
}

func wrapWriter(w io.Writer, envName string) io.Writer {
```